examples/step-60/step-60.cc

```diff
@@ -0,0 +1,1077 @@
+    // an absolution position.
+    //
+    // In the first case, the class MappingQEulerian offers its services, while
+    // in the second one, we'll use the class MappingFEField. They are in fact
+    // very similar. MappingQEulerian will only work for systems of FE_Q finite
+    // element spaces, where the displacement vector is stored in the first
+    // `spacedim` components of the FESystem, and the degree given as a
+    // parameter at construction time, must match the degree of the first
+    // `spacedim` components.
+    //
+    // The class MappingFEField is slightly more general, in that it allows you
+    // to select arbitrary FiniteElement types when constructing your
+    // approximation. Naturally some choices may (or may not) make sense,
+    // according to the type of FiniteElement you choose. MappingFEField
+    // implements the pure iso-parametric concept, and can be used, for example,
+    // to implement iso-geometric analysis codes in deal.II, by combining it
+    // with the FEBernstein finite element class. In this example, we'll use the
+    // two interchangeably, by taking into account the fact that one
+    // configuration will be a `displacement`, while the other will be an
+    // absolute `deformation` field.
+
+    if (parameters.use_displacement == true)
+      embedded_mapping =
+        std_cxx14::make_unique<MappingQEulerian<dim, Vector<double>, spacedim> >
+        (parameters.embedded_configuration_finite_element_degree,
+         *embedded_configuration_dh, embedded_configuration);
+    else
+      embedded_mapping =
+        std_cxx14::make_unique<MappingFEField<dim, spacedim, Vector<double>, DoFHandler<dim,spacedim> > >
+        (*embedded_configuration_dh,
+         embedded_configuration);
+
+    // Estimating the diameter of the largest active cell of $\Gamma$, and
+    // the smalles one of $\Omega$
+    double embedded_space_maximal_diameter =
+      GridTools::maximal_cell_diameter(*embedded_grid, *embedded_mapping);
+    double embedding_space_minimal_diameter =
+      GridTools::minimal_cell_diameter(*space_grid);
+    // Setting up the $\Gamma$'s DoFs
+    setup_embedded_dofs();
+
+    // With the mapping in place, it is now possible to query what is the
+    // location of all support points associated with the `embedded_dh`, by
+    // calling the method DoFTools::map_dofs_to_support_points.
+    //
+    // This method has two variants. One that does *not* take a Mapping, and
+    // one that takes a Mapping. If you use the second type, like we are doing
+    // in this case, the support points are computed through the specified mapping,
+    // which can manipulate them accordingly.
+    //
+    // This is precisely what the `embedded_mapping` is there for.
+
+    std::vector<Point<spacedim> > support_points(embedded_dh->n_dofs());
+    if (parameters.delta_refinement != 0)
+      DoFTools::map_dofs_to_support_points(*embedded_mapping,
```

Comment:
I don't remember what `delta_refinement` was for (which may suggest that the name is not well chosen), but why do you not need to build the map in some cases?